### PR TITLE
Added command cooldowns.

### PR DESCRIPTION
**Pull request by ggggg**

### CODE REVIEW

File: src/BPEssentials/Core.cs [hunk 1]
```diff
@@ -106,48 +111,60 @@ public void RegisterCommands()
...
-                            player.TS("command_failed_crimes", command.CommandName);
-                            return false;
-                        }
-                        if (!command.AllowWhileKO && player.IsKnockedOut)
+                        if (command.Disabled)
                         {
-                            player.TS("command_failed_crimes", command.CommandName);
+                            player.TS("command_disabled", command.CommandName);
                             return false;
                         }
```
Comment (PLASMAchicken) [inline src/BPEssentials/Core.cs:132]: Huh?

File: src/BPEssentials/Core.cs [hunk 2]
```diff
@@ -106,48 +111,60 @@ public void RegisterCommands()
...
-                        if (!command.AllowWhileCuffed && player.IsRestrained)
-                        {
-                            player.TS("command_failed_cuffed", command.CommandName);
-                            return false;
-                        }
-                        if (!command.AllowWhileJailed && player.job is Prisoner)
+
+                        if (!player.GetExtendedPlayer().EnabledBypass)
```
Comment (PLASMAchicken) [inline src/BPEssentials/Core.cs:128]:
```suggestion
                        if (player.GetExtendedPlayer().EnabledBypass)
                        {
                        return true;
                        }
```

File: src/BPEssentials/Core.cs [hunk 3]
```diff
@@ -106,48 +111,60 @@ public void RegisterCommands()
             foreach (var command in Settings.Commands)
             {
                 Logger.LogInfo($"[C] Registering command {command.CommandName}..");
-                if (!CommandInjection.TryGetCommandMethodDelegateByTypeName(command.CommandName, out var del, out var instance))
+                if (!CommandInjection.TryGetCommandMethodDelegateByTypeName(command.CommandName, out var del,
+                    out var instance))
```
Comment (PLASMAchicken) [inline src/BPEssentials/Core.cs:115]: Huh?

File: src/BPEssentials/Core.cs [hunk 4]
```diff
@@ -106,48 +111,60 @@ public void RegisterCommands()
...
+
+                            if (!command.AllowWithCrimes && player.wantedLevel != 0)
+                            {
+                                player.TS("command_failed_crimes", command.CommandName);
+                                return false;
+                            }
+
+                            if (command.CoolDown > 0 && CommandsCooldownHandler.IsCooldown(player.svPlayer, command.CommandName, command.CoolDown))
+                            {
+                                //TODO localtion
```
Comment (PLASMAchicken) [inline src/BPEssentials/Core.cs:156]: Huh?

**Review by PLASMAchicken — CHANGES_REQUESTED** [review]

File: src/BPEssentials/Core.cs (hunk 3 above)
Comment (ggggg) [reply src/BPEssentials/Core.cs:115]: What was even changed (but formating)?

Comment (PLASMAchicken) [reply src/BPEssentials/Core.cs:115]: Nothing but formatting wich is bad

File: src/BPEssentials/Core.cs (hunk 4 above)
Comment (ggggg) [reply src/BPEssentials/Core.cs:159]: wdym

File: src/BPEssentials/Core.cs (hunk 3 above)
Comment (ggggg) [reply src/BPEssentials/Core.cs:115]: how so

Comment (PLASMAchicken) [reply src/BPEssentials/Core.cs:115]: This linebreak is ugly

File: src/BPEssentials/Core.cs (hunk 4 above)
Comment (PLASMAchicken) [reply src/BPEssentials/Core.cs:159]: Todo comment ?!

**Review by PLASMAchicken — CHANGES_REQUESTED** [review]
.

File: src/BPEssentials/Core.cs (hunk 4 above)
Comment (ggggg) [reply src/BPEssentials/Core.cs:159]: ah

File: src/BPEssentials/Core.cs (hunk 1 above)
Comment (ggggg) [reply src/BPEssentials/Core.cs:132]: What?

**Comment by PLASMAchicken** [comment]
Failed to fix Suggested Changes and PR is stale.

**Pull request closed without merging**